### PR TITLE
Legal name updates for Entity Emailer

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#15532

*Description of changes:*
- Updated emailer to follow latest GCP listener structure
- Updated emailer project to use toml
- Business model references updated to LegalEntity
- `business.legal_name` references updated to use `business_name`
- Update docker file

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: queue_services/entity-emailer/src/entity_emailer/config.py
Comment (leodube-aot) [inline queue_services/entity-emailer/src/entity_emailer/config.py:59]: Should I run the Black formatter on all files in entity-emailer, or should I undo the double quotes?

File: queue_services/entity-emailer/src/entity_emailer/email_processors/__init__.py
Comment (leodube-aot) [inline queue_services/entity-emailer/src/entity_emailer/email_processors/__init__.py:36]: Should the variable be renamed to "entity" (or something similar)? Like: `entity = ` instead of `business = `

File: queue_services/entity-emailer/src/entity_emailer/config.py
Comment (leodube-aot) [inline queue_services/entity-emailer/src/entity_emailer/config.py:109]: I am not too familiar with Q topics. I set this based on the old `ENTITY_EVENT_PUBLISH_OPTIONS.subject` variable. Let me know if this is incorrect and I will fix here and the other couple places it is used.

File: queue_services/entity-emailer/pyproject.toml
Comment (argush3) [inline queue_services/entity-emailer/pyproject.toml:21]: I know we pull in the `jinja2` dependency via the `legal_api` module, but can we import it explicitly here?  

Just want to make that explicit for anything emailer actually uses.

Can you go through and see if there's any other dependency like that?

File: queue_services/entity-emailer/src/entity_emailer/email_processors/__init__.py
Comment (argush3) [reply queue_services/entity-emailer/src/entity_emailer/email_processors/__init__.py:36]: i think it's fine for now to leave it as `business`

File: queue_services/entity-emailer/src/entity_emailer/config.py
Comment (leodube-aot) [reply queue_services/entity-emailer/src/entity_emailer/config.py:74]: Sounds good, I'll hold off on running the formatter for now just to keep the files changed more legible for reviews. I'll run it through the black formatter as the last thing I do before merging.

File: queue_services/entity-emailer/pyproject.toml
Comment (leodube-aot) [reply queue_services/entity-emailer/pyproject.toml:21]: That makes sense, I'll have a look to see if there are any other dependencies that the emailer uses that are not explicitly imported.

**Comment by argush3** [comment]
I think you'll need to update some of the email templates referencing `business.legalName` to use `business.businessName`

image

File: queue_services/entity-emailer/src/entity_emailer/resources/worker.py
Comment (argush3) [inline queue_services/entity-emailer/src/entity_emailer/resources/worker.py:156]: I don't think we need to publish  to any queues from the emailer.  I know we had the `publish_event` function that published to the "ENTITY_EVENT_TOPIC" topic but it isnever used. 

I think we can remove this.

File: queue_services/entity-emailer/src/entity_emailer/resources/worker.py
Comment (argush3) [inline queue_services/entity-emailer/src/entity_emailer/resources/worker.py:159]: maybe add logging here to mimic the structure of entity-pay.  

i.e.
```
structured_log(request, "INFO", f"completed ce: {str(ce)}")
return {}, HTTPStatus.OK
```

File: queue_services/entity-emailer/src/entity_emailer/resources/worker.py
Comment (leodube-aot) [reply queue_services/entity-emailer/src/entity_emailer/resources/worker.py:156]: Do we want to keep the GCP Queue code after removing? This is the only place that something is published to a queue.

Comment (argush3) [reply queue_services/entity-emailer/src/entity_emailer/resources/worker.py:155]: Leave this code for now.  Maybe @thorwolpert or @vysakh-menon-aot have background on whether we eventually plan on publishing to the event topic or why it was there in the first place.

File: queue_services/entity-emailer/src/entity_emailer/config.py
Comment (leodube-aot) [reply queue_services/entity-emailer/src/entity_emailer/config.py]: Removed along with publish to queue

File: queue_services/entity-emailer/src/entity_emailer/resources/worker.py
Comment (argush3) [reply queue_services/entity-emailer/src/entity_emailer/resources/worker.py]: We can keep the GCP queue service code.  

I took a deeper look at worker.py code and we need it even if we don't publish a queue message.  I believe it's used at least to initialize the queue and read incoming queue messages.

**Review by argush3 — APPROVED** [review]
LGTM.  Nice work!

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**
